include/small/small.h

```diff
@@ -236,6 +236,15 @@ small_alloc_check(struct small_alloc *alloc)
 	return slab_cache_check(alloc->cache);
 }
 
+/**
+ * Return the size of the mempool object that is actually allocated for the
+ * requested size. Note that it can return different values for the same input,
+ * depending on the current `small_mempool->used_pool'.
+ * If the object is allocated on the large slab (by malloc), 0 is returned.
+ */
+size_t
+small_get_objsize(struct small_alloc *alloc, size_t size);
```

Comment:
I don't like that this function returns 0 for large allocations. It may be better to make this function return the actual allocation size and type for all kinds of allocations. For example,

```C
struct small_alloc_info {
        bool is_large;
        size_t real_size;
};

void
small_alloc_info(void *ptr, size_t size, struct small_alloc_info *info);
```

BTW for large allocations the real allocation size should probably be `size + slab_sizeof()`.

Reply:
Done.

We do not account the slab header for non-large allocations, I think it should be omitted for large allocations too, for consistency.